src/Raven.Server/Documents/Indexes/Persistence/Corax/QueryEnviroment.cs

```diff
@@ -0,0 +1,47 @@
+using System.Collections.Generic;
+using Corax;
+using Corax.Queries;
+using Raven.Server.Documents.Queries;
+using Raven.Server.ServerWide.Context;
+using Sparrow.Json;
+using RavenConstants = Raven.Client.Constants;
+using IndexSearcher = Corax.IndexSearcher;
+using CoraxConstants = Corax.Constants;
+using MoreLikeThisQuery = Raven.Server.Documents.Queries.MoreLikeThis.CoraxMoreLikeThisQuery;
+namespace Raven.Server.Documents.Indexes.Persistence.Corax;
+
+internal readonly struct QueryEnvironment
+{
```

Comment:
What purpose does this `QueryEnvironment` structure has? From what I see, unless this is a 1 time thingy, just the sheer size of it is telling me it should be a `class` and not a `struct`

Reply:
Moreover, IF we want to keep it a struct you must decorate every single parameter passed function with the `in` modifier to ensure that the content of this struct is not tracked at all and be treated as a constant.